Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -394,16 +394,20 @@ extension URL {
     }
 
     var pathHasTrailingSlash: Bool {
-        let url = self.absoluteString
+        if #available(OSX 10.11, iOS 9.0, tvOS 9.0, watchOS 2.0, *) {
+            return self.hasDirectoryPath
+        } else {
+            let url = self.absoluteString
 
-        var pathEndIndex = url.index(before: url.endIndex)
-        if let queryIndex = url.firstIndex(of: "?") {
-            pathEndIndex = url.index(before: queryIndex)
-        } else if let fragmentIndex = url.firstIndex(of: "#") {
-            pathEndIndex = url.index(before: fragmentIndex)
-        }
+            var pathEndIndex = url.index(before: url.endIndex)
+            if let queryIndex = url.firstIndex(of: "?") {
+                pathEndIndex = url.index(before: queryIndex)
+            } else if let fragmentIndex = url.suffix(from: url.firstIndex(of: "@") ?? url.startIndex).lastIndex(of: "#") {
+                pathEndIndex = url.index(before: fragmentIndex)
+            }
```

Comment:
Do we need the `else` branch? The minimum versions above were all released in 2015, making them 4 years old this year. Can we get away without it?

If the answer is really no, then can we also add a comment to this code branch to say that the code is not 100% correct, but is considered "good enough"?

Reply:
What do you mean by getting away without it? Adding supported platforms in `Package.swift` or a precondition? I would prefer not to have an alternative clause here as well...
@tomerd @tanner0101 @ianpartridge do you think we should limit the client to modern(er) platforms?

Reply:
given our main use case is linux, restricting older non-linux platforms in Package.swift seems fine to me